code/workspaces/web-app/src/components/app/ProjectSwitcher.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { withStyles } from '@material-ui/core';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import MenuItem from '@material-ui/core/MenuItem';
+import TextField from '@material-ui/core/TextField';
+import useCurrentProject from '../../hooks/useCurrentProject';
+import useProjectsArray from '../../hooks/useProjectsArray';
+import PromisedContentWrapper from '../common/PromisedContentWrapper';
+import projectsActions from '../../actions/projectActions';
+
+// Removing switcher margin and giving margin to promised content wrapper
+// make size constant when switching from circular progress to dropdown component
+const style = theme => ({
+  switcher: {
+    margin: 0,
+  },
+  promisedContent: {
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+    margin: `${theme.spacing(2)}px 0`,
+  },
+  dropdownProgress: {
+    width: '100%',
+    display: 'flex',
+    justifyContent: 'center',
+  },
+});
+
+export function createRoute(location, projectKey) {
+  const pattern = new RegExp('/projects/.*?/');
+  return location.pathname.replace(pattern, `/projects/${projectKey}/`);
```

Comment:
Why does this need a regex? Are you not able to do
```
location.pathname = `/projects/${projectKey}`;
```

Reply:
The idea is that it stays on the same page when you switch projects, so if you are on the notebooks page and switch project, you will be on the notebooks page for the new project. 

Using regex, I am able to effectively just update the project name in the current path so 
`/projects/oldproj/notebooks` goes to `/projects/newproj/notebooks`.